src/bundle.c

```diff
@@ -426,126 +433,157 @@ int remove_bundle(const char *bundle_name)
 		return ret;
 	}
 
-	/* os-core bundle not allowed to be removed...
-	 * although this is going to be caught later because of all files
-	 * being marked as 'duplicated' and note removing anything
-	 * anyways, better catch here and return success, no extra work to be done.
-	 */
-	if (strcmp(bundle_name, "os-core") == 0) {
-		ret = EBUNDLE_NOT_TRACKED;
-		goto out_free_curl;
-	}
-
-	if (!is_tracked_bundle(bundle_name)) {
-		ret = EBUNDLE_NOT_TRACKED;
-		fprintf(stderr, "Warning: Bundle \"%s\" does not seem to be installed\n", bundle_name);
-		goto out_free_curl;
-	}
-
 	current_version = get_current_version(path_prefix);
 	if (current_version < 0) {
 		fprintf(stderr, "Error: Unable to determine current OS version\n");
 		ret = ECURRENT_VERSION;
-		goto out_free_curl;
+		telemetry(TELEMETRY_CRIT,
+			  "bundleremove",
+			  "bundle=%s\n"
+			  "current_version=%d\n"
+			  "result=%d\n",
+			  *bundles,
+			  current_version,
+			  ret);
+		swupd_deinit(lock_fd, &subs);
+		return ret;
 	}
 
 	mix_exists = (check_mix_exists() & system_on_mix());
 
-	current_mom = load_mom(current_version, false, mix_exists);
-	if (!current_mom) {
-		fprintf(stderr, "Unable to download/verify %d Manifest.MoM\n", current_version);
-		ret = EMOM_NOTFOUND;
-		goto out_free_curl;
-	}
+	for (; *bundles; ++bundles, total++) {
 
-	if (!search_bundle_in_manifest(current_mom, bundle_name)) {
-		fprintf(stderr, "Bundle name is invalid, aborting removal\n");
-		ret = EBUNDLE_REMOVE;
-		goto out_free_mom;
-	}
+		fprintf(stderr, "Removing bundle: %s\n", *bundles);
```

Comment:
This print ought to be after the check it is not os-core.

Reply:
Ok